src/derivative.c

```diff
@@ -244,117 +266,59 @@ OSQPInt adjoint_derivative(OSQPSolver*    solver,
     OSQPVectorf_free(rhs_temp2);
     OSQPVectorf_free(zeros);
 
-    // ----------- Check
-    adjoint_derivative_linsys_solver(solver, settings, P_full, G, A_eq, GDiagLambda, slacks, rhs);
+    // --------- stuff things into derivative_data
+    // All this should go inside _compute
+    derivative_data->rhs = OSQPVectorf_copy_new(rhs);
+    OSQPVectorf_free(rhs);
+
+    OSQPMatrix* P_full = OSQPMatrix_triu_to_symm(P);
+    OSQPMatrix_free(P);
+    adjoint_derivative_linsys_solver(solver, solver->settings, P_full, G, A_eq, GDiagLambda, slacks, derivative_data->rhs);
+    OSQPMatrix_free(P_full);
+    OSQPMatrix_free(G);
+    OSQPMatrix_free(A_eq);
+    OSQPMatrix_free(GDiagLambda);
+    OSQPVectorf_free(slacks);
 
-    OSQPFloat* rhs_data = OSQPVectorf_data(rhs);
+    OSQPFloat* rhs_data = OSQPVectorf_data(derivative_data->rhs);
 
     OSQPFloat* r_yl = (OSQPFloat *) c_malloc(m * sizeof(OSQPFloat));
     OSQPFloat* r_yu = (OSQPFloat *) c_malloc(m * sizeof(OSQPFloat));
     // TODO: We shouldn't have to do this if we assemble r_yl/r_yu judiciously
     for (j=0; j<m; j++) r_yl[j] = 0;
     for (j=0; j<m; j++) r_yu[j] = 0;
 
-    OSQPInt pos = n + n_ineq_l + n_ineq_u + n_eq;
-    OSQPVectorf* rx = OSQPVectorf_view(rhs, pos, n);
-
+    OSQPInt pos = n + derivative_data->n_ineq_l + derivative_data->n_ineq_u + derivative_data->n_eq;
     pos += n;
-    for (j=0; j<n_ineq_l; j++) {
+
+    for (j=0; j<derivative_data->n_ineq_l; j++) {
         r_yl[l_noninf_indices_vec[j]] = -rhs_data[pos+j];
     }
-    pos += n_ineq_l;
-    for (j=0; j<n_ineq_u; j++) {
+    c_free(l_noninf_indices_vec);
+    pos += derivative_data->n_ineq_l;
+
+    for (j=0; j<derivative_data->n_ineq_u; j++) {
         r_yu[u_noninf_indices_vec[j]] = rhs_data[pos+j];
     }
-    pos += n_ineq_u;
-    for (j=0; j<n_eq; j++) {
+    c_free(u_noninf_indices_vec);
+    pos += derivative_data->n_ineq_u;
+
+    for (j=0; j<derivative_data->n_eq; j++) {
         if (nu_sign_vec[j]==1) {
             r_yl[eq_indices_vec[j]] = 0;
             r_yu[eq_indices_vec[j]] = rhs_data[pos+j] / y_data[eq_indices_vec[j]];
         } else {
             r_yl[eq_indices_vec[j]] = -rhs_data[pos+j] / y_data[eq_indices_vec[j]];
             r_yu[eq_indices_vec[j]] = 0;
         }
-
     }
+    c_free(nu_sign_vec);
 
-    OSQPVectorf *ryl = OSQPVectorf_new(r_yl, m);
+    derivative_data->ryl = OSQPVectorf_new(r_yl, m);
     c_free(r_yl);
-    OSQPVectorf_ew_prod(ryl, ryl, y_l);
-    OSQPVectorf_mult_scalar(ryl, -1);
-    OSQPVectorf *ryu = OSQPVectorf_new(r_yu, m);
+    OSQPVectorf_ew_prod(derivative_data->ryl, derivative_data->ryl, derivative_data->y_l);
+    OSQPVectorf_mult_scalar(derivative_data->ryl, -1);
+    derivative_data->ryu = OSQPVectorf_new(r_yu, m);
```

Comment:
Allocate `derivative_data->ryu` in `osqp_setup` and just overwrite the data in that vector with the new values here.

Reply:
Done